### ISSUE TITLE
How to setup minSize, maxSize, defaultSize dynamic.

### ISSUE THREAD

**Issue by keval-key** [bvaughn/react-resizable-panels#401]
How to setup minSize, defaultSize dynamic.
I may don't know proper issue but you check demo by below link.

### How to Reproduce issue?
- Open Link: [https://codesandbox.io/p/sandbox/56q8ws](https://codesandbox.io/p/sandbox/56q8ws)
- 1st open left panel
- 2nd open right panel
- 3rd close left panel
- 4th open left panel and then you may see error.
### Error:
```
Error
Previous layout not found for panel index 2
```
### Video:
https://github.com/user-attachments/assets/02bc57d3-880e-46f9-b848-95ee794e7dda

**Comment by keval-key** [comment]
Yes, @bvaughn. But is there any way to make it dynamic? The reason for using dynamic elements is that I want to make it responsive.

**Comment by bvaughn** [comment]
What you're doing _should_ work. I don't know why it doesn't. Probably a bug in this library.

**Comment by keval-key** [comment]
Oh I see

**Comment by keval-key** [comment]
If you can help solve this issue, it would be greatly appreciated.

**Comment by Anuovec** [comment]
I encountered the same issue, I tried looking into it. It seems this code is causing the issue: https://github.com/bvaughn/react-resizable-panels/blob/main/packages/react-resizable-panels/src/Panel.ts#L183
Without it, it works as expected.

The comment says that it is only for pixel-based constants, is it needed in some cases?

**Comment by bvaughn** [comment]
> The comment says that it is only for pixel-based constants, is it needed in some cases?

That's a scenario in which I think the check is useful, but not the only one. I suspect this is a symptom of the bug but...the underlying bug is maybe somewhere else?

**Comment by Anuovec** [comment]
It seems the issue only occurs, when constraints are changed for the middle panel and at the same time a new left panel is being added.

**Comment by bvaughn** [comment]
> It seems the issue only occurs, when constraints are changed for the middle panel and at the same time a new left panel is being added.

I wonder if changing the panel id in this scenario might work around the bug?

**Comment by igortas** [comment]
@bvaughn I like also more examples with dynamic values, I tend to use this library in a way where, I have one fully expanded resizable panel, after that I can open one or two more panels dynamically and I can toggle them on and off, or hide the completely, so playing around with dynamic values it's bit problematic, is there a way, where values works out of the box and I can just provide how much panels I like at any given time(including toggled on/off panel)

**Comment by igortas** [comment]
@bvaughn stil this one is opened, so I like to point out, issues with toggling split panes. After I initially set som panes, and I like one of the panes to be still visible, but toggled (make smaller width of the pane when is in toggled state), nothing happens, even I'm setting some of the options, min, max, default... Workng with the panes in dynamic environment, is not easy and also laggish....

**Comment by keval-key** [comment]
> > It seems the issue only occurs, when constraints are changed for the middle panel and at the same time a new left panel is being added.
> 
> I wonder if changing the panel id in this scenario might work around the bug?

Oh yes, dynamic ID is a workaround solution for this.

**Comment by bvaughn** [comment]
I've just finished the next major release of this library, version 4.0. (More information available [here](https://github.com/bvaughn/react-resizable-panels/blob/main/CHANGELOG.md#4.0.0).)

I am closing **all open GitHub issues** as of this new release. If you've reported a bug that is still present in version 4, please **update the issue description** to add a Code Sandbox repro with react-resizable-panels version 4 and then re-open the issue. ([Here](https://codesandbox.io/p/sandbox/n9vs2g) is a sample sanbodx you can fork.)

> [!NOTE]
> Issues that are re-opened without a repro will be closed because I just don't have the time to try and chase down repros myself. Thanks for your understanding! 🙇🏼
